### ISSUE TITLE
brcmegl deprecated on Raspberry Pi 4?

### ISSUE THREAD

**Issue by bjanders** [libsdl-org/SDL#11444]
I was trying to compile SDL3 for a Raspberry Pi 4. In cmake/sdlchecks.cmake `macro(CheckRPI)` there is the condition:

```
if(TARGET PkgConfig::PC_BCM_HOST AND TARGET PkgConfig::PC_BRCMEGL)
```

However, it seems that the brcmegl library is not present on the Raspberry Pi 4.

I have  libudev-dev libasound2-dev libdbus-1-dev libraspberrypi0 libraspberrypi-bin libraspberrypi-dev installed as documented in https://github.com/libsdl-org/SDL/blob/main/docs/README-raspberrypi.md. /opt/vc does not exist on the device.

HAVE_RPI  would then not be defined.

This page suggest that brcmegl is deprecated on the Raspberry Pi 4: https://gitlab.freedesktop.org/gstreamer/gst-omx/-/issues/40:

> The example should be linked to brcmegl library for RPi target which is deprecated and not properly functioning on RPi4.

Should the check for brcmegl be removed (for RPi 4)? Or does  HAVE_PRI not have to be defined for Raspberry Pi 4?

**Comment by bjanders** [comment]
With a quick test, It seems to be working as intended. SDL and the examples build, and the examples succesfully run on a RPi 4, even though `HAVE_RPI` is not defined. So maybe just README-raspberrypi.md could be updated with regards to

> You also need the VideoCore binary stuff that ships in /opt/vc for EGL and OpenGL ES 2.x,

and mention that these are not required for the RPi 4 (and presumably RPi 5).

**Comment by cmitu** [comment]
The (Broadcom) EGL drivers and the DispmanX stack have been deemed legacy starting with the previous RaspiOS release (which was based on Debian 11 Bullseye), when the DRM/KMS drivers replaced them for all Pi models.

The Pi4 never supported the legacy GL drivers (though support for DispmanX was still present when using the `fkms` driver) and the Pi5 also. The current RaspiOS (based on Debian 12 bookworm) has made the full switch by removing any `/opt/vc` remnants.

That being said, one can still manually download the old drivers/firmware files from the RPI repos and compile the `raspberrypi` driver(s), but indeed the readme should specify that the driver(s) work only on RaspiOS 'buster' or older.

[1] https://www.raspberrypi.com/news/raspberry-pi-os-debian-bullseye/

**Comment by icculus** [comment]
As others said, this is checking the legacy video interfaces that newer Raspberry Pi OS has dropped. Kmsdrm works great with SDL3 on all modern RPi devices that just want to draw directly to the hardware and don't want the overhead of X11 or Wayland.

**Comment by cmitu** [comment]
How about adding to the a blurb for users to know that compilation will not work in current RaspiOS ?

``` diff
diff --git a/docs/README-raspberrypi.md b/docs/README-raspberrypi.md
index 9984ec6d1..fa626ca64 100644
--- a/docs/README-raspberrypi.md
+++ b/docs/README-raspberrypi.md
@@ -1,9 +1,16 @@
 Raspberry Pi
 ============

+Deprecation Notice
+---------
+
+The following information details the method to use the legacy Broadcom GL video/render stack that was available in Raspbian until the Buster release. Starting with RaspiOS 11 'bullseye', it has been deprecated by the Raspberry Pi engineers in favor of the open source Mesa VC4 driver, to be fully removed in RaspiOS 12 'bookworm'.
+
+Using the KMS video driver with the associated render(ers) can be used instead, having feature parity with the previous legacy closed driver and also supporting newer hardware (Pi4 and later), 64bit architectures and Vulkan renderer support.
+
 Requirements:

-Raspbian (other Linux distros may work as well).
+Raspbian Buster or later (other Linux distros may work as well).

```

**Comment by bjanders** [comment]
> How about adding to the a blurb for users to know that compilation will not work in current RaspiOS ?
 
Maybe I'm misunderstanding, but compilation works fine in current Raspberry Pi OS. I haven't tested with X11 or Wayland, though, if that is the issue, as I understand from @icculus message.

It would be helpful is the README-raspberrypi.md would clarify what works and what does not work, and that it is correct that `s HAVE_PRI` is not defined for newer Raspberry Pi OS. This is what got me initially trying to solve an issue that didn't exist.

To summarize, I think the README-raspberrypi.md needs two parts, highlighting that there two different methods depending on the OS version used: How to compile on bookworm and newer, and how to compile using the legacy method.

**Comment by icculus** [comment]
It works, it just won't use the obsolete video driver, instead using kmsdrm, the new video driver that is a standard Linux interface instead of a raspberry pi specific one.

I can clarify the docs a little.
